sdk/core/azure-core/src/test/java/com/azure/core/util/FluxUtilTest.java

```diff
@@ -211,6 +211,39 @@ public void testWriteFileWithPosition() throws Exception {
         }
     }
 
+    @Test
+    public void testWriteFileWithReSubscription() throws Exception {
+        String initialContent = "hello there";
+        String firstWrite = "test";
+        String secondWrite = "again";
+        String expectedContent = "hello testagain";
+
+        byte[] firstBytes = firstWrite.getBytes(StandardCharsets.UTF_8);
+        byte[] secondBytes = secondWrite.getBytes(StandardCharsets.UTF_8);
+
+        Flux<ByteBuffer> firstBody = Flux.just(ByteBuffer.wrap(firstBytes));
+        Flux<ByteBuffer> secondBody = Flux.just(ByteBuffer.wrap(secondBytes));
+
+        File file = createFileIfNotExist();
+
+        try (FileOutputStream stream = new FileOutputStream(file)) {
+            stream.write(initialContent.getBytes(StandardCharsets.UTF_8));
+        }
+
+        Mono.fromCallable(() -> AsynchronousFileChannel.open(file.toPath(), StandardOpenOption.WRITE))
```

Comment:
I think we should change this logic to be what was found to cause the problem in the first place. With having the Reactor retry and a data generator.

Something like:

- Have a `Flux<ByteBuffer>` where on the first subscription it writes `"Hello "`
- Then emit and error as part of the `Flux<ByteBuffer>` which triggers the Reactor retry
- Then have the second subscription to the `Flux<ByteBuffer>` write the expected bytes

And maybe increase that two having two error and the third subscription writing the content correctly for extra validation.